### ISSUE TITLE
Cryptic error on disk full

### ISSUE THREAD

**Issue by carlosmn** [libgit2/libgit2#2965]
When the disk is full, instead of returning an error saying that, we get a rather non-descriptive error message about a particular function saying there was some error. We should probably let the callback set the error or read it from `errno` so we can actually tell the user something useful.

```
Traceback (most recent call last):
  File "test_fmc.py", line 31, in setUp
    bare=True, checkout_branch="foo_7.0_integration")
  File "/Users/carlos/foo/tools/venv/lib/python2.7/site-packages/pygit2-0.22.0-py2.7-macosx-10.9-x86_64.egg/pygit2/__init__.py", line 286, in clone_repository
    check_error(err)
  File "/Users/carlos/foo/tools/venv/lib/python2.7/site-packages/pygit2-0.22.0-py2.7-macosx-10.9-x86_64.egg/pygit2/errors.py", line 64, in check_error
    raise GitError(message)
GitError: git_path_direach callback returned -1
```

**Comment by pypingou** [comment]
I am seeing the same error when trying to clone a git repo into a temp directory but in my case the disk isn't full.

Any clue what is wrong?

**Comment by ethomson** [comment]
@pypingou Are you cloning a local repository?

**Comment by pypingou** [comment]
@ethomson yes I am

**Comment by ethomson** [comment]
This sounds like you're running into #2797.  You can work around this issue by not trying to link, but always copy.  See https://libgit2.github.com/libgit2/#HEAD/type/git_clone_local_t for details, but setting `git_clone_options.local` to `GIT_CLONE_LOCAL_NO_LINKS` (for example) should avoid this.

**Comment by pypingou** [comment]
Thanks for the quick feedback!

I'm running into this via pygit2, do you know if it supports this?

**Comment by pypingou** [comment]
ok I went around the problem by moving the place of the tmp folder where I do the clone into the same mount point (/srv in my case) as where the original git repo is.
ugly but works for now :)

Thanks for your help!

**Comment by carlosmn** [comment]
Just because you're cloning into a temp dir it doesn't mean we fail. That should work perfectly fine, but unfortunately we hide the error message. If you can strace your program, you should be able to see the error which the OS returns. We will most likely backport the fix in #3132 to 0.22 but meanwhile there's no good way to know *why* there was an error. #2797 is a not that common a setup.
